File: 2trabajoConstructores/c#/claseNormal/AnimalCopia.cs
public class AnimalCopia
{
    public string Nombre { get; set; }
    public string Sonido { get; set; }

    public AnimalCopia(string nombre, string sonido)
    {
        Nombre = nombre;
        Sonido = sonido;
    }

    // Constructor copia
    public AnimalCopia(AnimalConParametros original)
    {
        Nombre = original.Nombre;
        Sonido = original.Sonido;
    }

    public void HacerSonido()
    {
        Console.WriteLine($"{Nombre} hace: {Sonido}");
    }
}

File: 2trabajoConstructores/go/claseNormal/animal_copia.go
package main

import "fmt"

type AnimalCopia struct {
	Nombre string
	Sonido string
}

func NuevoAnimalCopia(original AnimalConParametros) AnimalCopia {
	return AnimalCopia{
		Nombre: original.Nombre,
		Sonido: original.Sonido,
	}
}

func (a AnimalCopia) HacerSonido() {
	fmt.Printf("COPIA → %s hace: %s\n", a.Nombre, a.Sonido)
}
